core/models/identity.py

```diff
@@ -157,6 +157,9 @@ def create_identity(cls, username, provider_location,
             c_key = c_key.replace('cred_', '')
             credentials[c_key] = c_value
 
+        if 'router_name' not in credentials:
+            credentials['router_name'] = provider.select_router()
+
```

Comment:
This line is supposed to ensure that all ​*new*​ identities are created with a `router_name` at the identity level.

Currently, when `service/accounts/openstack.py` calls create_identity, it does not explicitly create a router name.

The ​*old*​ behavior was to define the router name at the ​*provider*​ level.
The ​*new*​ behavior will defer to the old behavior for 'old' identities, but any identity that has a `router_name` assigned will override the value.